src/coreclr/debug/daccess/daccess.cpp

```diff
@@ -4766,12 +4766,15 @@ ClrDataAccess::SetAllTypeNotifications(
 HRESULT STDMETHODCALLTYPE
 ClrDataAccess::SetAllCodeNotifications(
     /* [in] */ IXCLRDataModule* mod,
-    /* [in] */ ULONG32 flags)
+    /* [in] */ ULONG32 flagsRaw)
 {
     HRESULT status;
 
     DAC_ENTER();
 
+    _ASSERTE(flagsRaw <= USHRT_MAX);
```

Comment:
This assert is not generally valid. It is valid only after `IsValidMethodCodeNotification` has been called.

Reply:
I'm not sure why. All uses of the `flags` argument are implicitly down cast to `USHORT`.

Reply:
This is public method that is presumably called by code outside this repo. It is trying to do argument validation and return `E_INVALIDARG` when the arguments are wrong. 

If the method is called with `flags: 0x10001`, it should return `E_INVALIDARG` and not assert.

Reply:
I see. It won't do that right now because `IsValidMethodCodeNotification(flags)` implicitly downcasts to `USHORT`. I will update to do as suggested.